sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -230,9 +231,23 @@ case class Union(children: Seq[LogicalPlan]) extends LogicalPlan {
   }
 
   // updating nullability to make all the children consistent
-  override def output: Seq[Attribute] =
-    children.map(_.output).transpose.map(attrs =>
-      attrs.head.withNullability(attrs.exists(_.nullable)))
+  override def output: Seq[Attribute] = {
```

Comment:
Don't we need this logic in UnionExec too?

Reply:
I'd say so, but I have not yet been able to find a use case and therefore a UT for that too. If you check the failing case of the JIRA, for instance, the current change works. Because after the analysis all the plans children of union have their attributes casted appropriately. So I am not sure that is really needed.

Reply:
I still feel a bit weird about different output between `Union`/`UnionExec`... (But, I don't have any better solution...)

Reply:
+1, it would be better to keep the output consistent between logical and phyiscal plan, although there is no direct benefit.